### PR TITLE
Update documentation

### PR DESCRIPTION
**Pull request by andersy005**
In preparation for a new release, this PR updates the documentation to reflect some of the changes made in #194

### CODE REVIEW

File: docs/source/notebooks/tutorial.ipynb [hunk 1]
```diff
@@ -4,40 +4,30 @@
    "cell_type": "markdown",
    "metadata": {},
    "source": [
-    "# Tutorial"
+    "# Tutorial\n",
+    "\n",
+    "Computer simulations of the Earth’s climate and weather generate huge amounts of data. These data are often persisted on HPC systems or in the cloud across multiple data assets of a variety of formats (netCDF, zarr, etc…). Finding, investigating, loading these data assets into compute-ready data containers costs time and effort. The data user needs to know what data sets are available, the attributes describing each data set, before loading a specific data set and analyzing it. Intake ESM is a Python package that aims to provide a higher-level interface to searching and loading Earth System Model (ESM) data archives, such as CMIP6. "
```
Comment (bonnland) [inline docs/source/notebooks/tutorial.ipynb:9]: "Computer simulations of the Earth’s climate and weather generate huge amounts of data. These data are often persisted on HPC systems or in the cloud across multiple data assets in a variety of formats (netCDF, zarr, etc…). Finding, investigating, and loading these data assets costs time and effort. The end user needs to not only know what data sets are available, but also the attributes describing each data set, before loading a specific data set and analyzing it.  The intake-esm package aims to provide a simple, high-level interface to searching and loading Earth System Model (ESM) data archives, such as CMIP6. "

Comment (bonnland) [reply docs/source/notebooks/tutorial.ipynb:9]: Just small suggestions, all subjective.

File: docs/source/notebooks/tutorial.ipynb [hunk 2]
```diff
@@ -4,40 +4,30 @@
...
-    "For demonstration purposes, we will be using the CMIP6 data hosted in Google Cloud Storage. For this data collection, we will be using the Pangeo cloud collection file stored [here](https://raw.githubusercontent.com/NCAR/intake-esm-datastore/master/catalogs/pangeo-cmip6.json).\n",
+    "Intake ESM works as follows:\n",
     "\n",
-    "    \n",
     "\n",
-    "More collection catalogs are listed [here](https://intake-esm.readthedocs.io/en/latest/faq.html#is-there-a-list-of-existing-catalogs). \n",
-    "    \n",
-    "    \n",
-    "</div>\n"
+    "1. Parse an ESM data catalog (stored in a JSON file). The JSON file must conform to the [ESM Collection Specification](https://github.com/NCAR/esm-collection-spec).\n",
```
Comment (bonnland) [inline docs/source/notebooks/tutorial.ipynb:21]: "1. Load an ESM data catalog file. The JSON-based catalog file must conform to the [ESM Collection Specification](https://github.com/NCAR/esm-collection-spec).\n",

File: docs/source/notebooks/tutorial.ipynb [hunk 3]
```diff
@@ -278,30 +272,156 @@
...
+   ],
+   "source": [
+    "len(col.keys())"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Intake ESM uses information saved in the `key_template` property in order to figure out how to create catalog entry keys:"
```
Comment (bonnland) [inline docs/source/notebooks/tutorial.ipynb:302]: Intake ESM uses the `key_template` property in order to create catalog entry keys

File: docs/source/notebooks/tutorial.ipynb [hunk 4]
```diff
@@ -278,30 +272,156 @@
...
    "source": [
-    "col.df.columns"
+    "col.keys()[:10]"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Each key in this list points to all assets (files) that will be concatenated, merged into a single, high-level aggregated xarray dataset."
```
Comment (bonnland) [inline docs/source/notebooks/tutorial.ipynb:362]: "Each key in this list points to the assets (files) that will be merged into a single xarray dataset."

File: docs/source/notebooks/tutorial.ipynb [hunk 5]
```diff
@@ -407,24 +531,21 @@
    "source": [
     "### Searching for specific datasets\n",
     "\n",
-    "\n",
-    "One of the features supported in ``intake-esm`` is querying the collection catalog.\n",
-    "This feature is provided via the ``search()`` method. The ``search()`` method allows the user to\n",
-    "specify a query by using keyword arguments. This method returns a subset of the collection catalog\n",
+    "The ``search()`` method allows the user to specify a query by using keyword arguments. This method returns a subset of the catalog\n",
```
Comment (bonnland) [inline docs/source/notebooks/tutorial.ipynb:534]: "query" instead of "specify a query by"

File: docs/source/notebooks/tutorial.ipynb [hunk 6]
```diff
@@ -752,14 +892,14 @@
    "source": [
     "As you can see, the search results above include `source_ids` for which we only have one of the two variables, and one or two of the three experiments.\n",
     "\n",
-    "We can tell `intake-esm` to discard any `source_id` that doesn't have both variables `[\"thetao\", \"o2\"]` and all three experiments `[\"historical\", \"ssp245\", \"ssp585\"]` by passing `require_all_on=[\"source_id\"]` to the search method:\n",
+    "We can tell intake ESM to discard any `source_id` that doesn't have both variables `[\"thetao\", \"o2\"]` and all three experiments `[\"historical\", \"ssp245\", \"ssp585\"]` by passing `require_all_on=[\"source_id\"]` to the search method:\n",
```
Comment (bonnland) [inline docs/source/notebooks/tutorial.ipynb:895]: Is "intake ESM" intended here?   I thought maybe the original was more accurate?

File: docs/source/notebooks/tutorial.ipynb [hunk 7]
```diff
@@ -834,52 +974,26 @@
...
-    "\n",
-    "\n",
-    "\n",
-    "Note that \n",
-    "- the data could be stored in the cloud or on a local filesystem, \n",
-    "- the data could be stored in [zarr](https://zarr.readthedocs.io/en/stable/) format, [netCDF](https://www.unidata.ucar.edu/software/netcdf/) format. \n",
+    "## Loading datasets\n",
     "\n",
-    "**And all of this is opaque to the user!**\n",
+    "As pointed earlier, intake ESM contains special logic to figure out how to concatenate and merge the individual results of your query into larger, more high-level aggregated [xarray datasets](http://xarray.pydata.org/en/stable/api.html#dataset).\n",
```
Comment (bonnland) [inline docs/source/notebooks/tutorial.ipynb:979]: "As pointed out earlier, intake-esm contains logic to merge the query results into higher level [xarray datasets]"

Comment (bonnland) [reply docs/source/notebooks/tutorial.ipynb:979]: Although my background may bias me, I think most users will already have some intuition for the benefits of xarray objects over individual files.

**Comment by jhamman** [comment]
Looks great, beyond @bonnland's comments, I just want to add one more. I think you should correct the heading `Using to_dataset()` -> `Using to_dataset_dict()`.

**Comment by andersy005** [comment]
Thank you @bonnland & @jhamman! I addressed all issues/comments.

**Comment by bonnland** [comment]
@andersy005   Everything looks fantastic, outside of the final note at the end of the tutorial:  The note is not rendering link markdown correctly. 
 
Also, please double check that this link is publicly accessible because I cannot access it: 
 https://github.com/jbusecke/cmip6_preprocessing/blob/master/notebooks/tutorial_intake_esm_preprocessing.ipynb

**Comment by andersy005** [comment]
Thank you @bonnland! I fixed the cmip6-preprocessing package URL.

**Pull request merged**